libreyolo/export/tensorrt.py

```
@@ -338,6 +338,38 @@ def export_tensorrt(
+            if half:
+                # ViT backbones (DINOv2/v3) overflow in FP16 -> NaN. Detect a ViT
+                # backbone (LayerNorm/Erf under model/backbone) and pin its float
+                # compute layers to FP32 (mixed precision); no-op for CNN backbones.
+                try:
+                    import onnx as _onnx
+                    _vit = any(
+                        n.op_type in ("LayerNormalization", "Erf")
+                        and "backbone" in (n.name or "")
+                        for n in _onnx.load(onnx_path).graph.node
+                    )
+                except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)